src/stream-tcp-reassemble.c

```diff
@@ -1846,7 +1846,11 @@ int StreamTcpReassembleHandleSegment(ThreadVars *tv, TcpReassemblyThreadCtx *ra_
     } else if (p->tcph->th_flags & TH_RST) { // accepted rst
         dir = UPDATE_DIR_PACKET;
     } else if ((p->tcph->th_flags & TH_FIN) && ssn->state > TCP_TIME_WAIT) {
-        dir = UPDATE_DIR_PACKET;
+	if (p->tcph->th_flags & TH_ACK) {
+	    dir = UPDATE_DIR_BOTH;
+	} else {
+            dir = UPDATE_DIR_PACKET;
+	}
```

Comment:
Hi @amirabell !
formatting seems to be off here. Could you please check?

Reply:
I am sorry, I didn't notice. Opened a new PR: https://github.com/OISF/suricata/pull/6693